.github/workflows/build.yml

```diff
@@ -174,6 +174,26 @@ jobs:
         with:
           path: ${{ env.CACHED_BUILD_PATHS }}
           key: ${{ env.BUILD_CACHE_KEY }}
+
+      - name: NX cache
+        uses: actions/cache@v3
+        # Disable cache when:
+        # - on master
+        # - on release branches
+        # - when PR has `ci-skip-cache` label
+        if:
+          github.ref != 'refs/heads/master' && !startsWith(github.ref, 'refs/heads/release/') &&
+          !github.event.pull_request.labels.contains('ci-skip-cache')
```

Comment:
The idea is you can add a label `ci-skip-cache` to the PR, which ensures it will not do any nx caching. May be helpfup when debugging here or there (maybe we could even extend this to the other caches, like the dependency cache?)